### PR TITLE
CD doc fix from generator utility

### PR DESCRIPTION
**Pull request by gmarjoram**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Relates OR Closes #0000

Output from acceptance testing:

```
$ make testacc TESTARGS='-run=TestAccXXX'

...
```

### CODE REVIEW

**Comment by gmarjoram** [comment]
Acceptance tests ->
```
make testacc TEST=./ibm/service/cdtoolchain TESTARGS=‘-run=TestAccIBMCdToolchain*’
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./ibm/service/cdtoolchain -v ‘-run=TestAccIBMCdToolchain*’ -timeout 700m
=== RUN   TestAccIBMCdToolchainDataSourceBasic
--- PASS: TestAccIBMCdToolchainDataSourceBasic (21.63s)
=== RUN   TestAccIBMCdToolchainDataSourceAllArgs
--- PASS: TestAccIBMCdToolchainDataSourceAllArgs (17.00s)
=== RUN   TestAccIBMCdToolchainToolAppconfigDataSourceBasic
--- PASS: TestAccIBMCdToolchainToolAppconfigDataSourceBasic (18.07s)
=== RUN   TestAccIBMCdToolchainToolAppconfigDataSourceAllArgs
--- PASS: TestAccIBMCdToolchainToolAppconfigDataSourceAllArgs (18.88s)
=== RUN   TestAccIBMCdToolchainToolArtifactoryDataSourceBasic
--- PASS: TestAccIBMCdToolchainToolArtifactoryDataSourceBasic (17.56s)
=== RUN   TestAccIBMCdToolchainToolArtifactoryDataSourceAllArgs
--- PASS: TestAccIBMCdToolchainToolArtifactoryDataSourceAllArgs (17.54s)
=== RUN   TestAccIBMCdToolchainBasic
--- PASS: TestAccIBMCdToolchainBasic (21.28s)
=== RUN   TestAccIBMCdToolchainAllArgs
--- PASS: TestAccIBMCdToolchainAllArgs (24.41s)
=== RUN   TestAccIBMCdToolchainToolPipelineBasic
--- PASS: TestAccIBMCdToolchainToolPipelineBasic (18.88s)
=== RUN   TestAccIBMCdToolchainToolPipelineAllArgs
--- PASS: TestAccIBMCdToolchainToolPipelineAllArgs (30.68s)
PASS
ok  	github.com/IBM-Cloud/terraform-provider-ibm/ibm/service/cdtoolchain	207.000s
```
```
make testacc TEST=./ibm/service/cdtektonpipeline TESTARGS=‘-run=TestAccIBMTekton*’
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./ibm/service/cdtektonpipeline -v ‘-run=TestAccIBMTekton*’ -timeout 700m
=== RUN   TestAccIBMTektonPipelineBasic
--- PASS: TestAccIBMTektonPipelineBasic (30.91s)
PASS
ok  	github.com/IBM-Cloud/terraform-provider-ibm/ibm/service/cdtektonpipeline	31.890s
```

**Comment by gmarjoram** [comment]
@hkantare looks like the SDK generator team has fixed the "computed variables having Required/Optional" problem you raised in my previous PR https://github.com/IBM-Cloud/terraform-provider-ibm/pull/3851. This PR is the result of running the new generator (3.52.0)

File: ibm/service/cdtoolchain/resource_ibm_cd_toolchain.go [hunk 1]
```diff
@@ -91,7 +91,7 @@ func ResourceIBMCdToolchain() *schema.Resource {
 }
 
 func ResourceIBMCdToolchainValidator() *validate.ResourceValidator {
-	validateSchema := make([]validate.ValidateSchema, 0)
+	validateSchema := make([]validate.ValidateSchema, 1)
```
Comment (hkantare) [inline ibm/service/cdtoolchain/resource_ibm_cd_toolchain.go:94]: This is the generator issue actually the length should be zero...we raised issue with generator team..can you revert these changes
